demos/app/iot_demo.c

```diff
@@ -118,19 +120,51 @@ int main( int argc,
     /* For a secured connection, set the members of the credentials. */
     if( demoArguments.securedConnection == true )
     {
-        /* Set credential paths. */
+        /* Set credential information. */
         credentials.pClientCert = demoArguments.pClientCertPath;
         credentials.pPrivateKey = demoArguments.pPrivateKeyPath;
         credentials.pRootCa = demoArguments.pRootCaPath;
+        credentials.pUserName = NULL;
+        credentials.pPassword = NULL;
+
+        /* Set the MQTT username, as long as it's not empty or NULL. */
+        if( demoArguments.pUserName != NULL )
+        {
+            credentials.userNameSize = strlen( demoArguments.pUserName );
+
+            if( credentials.userNameSize > 0 )
+            {
+                credentials.pUserName = demoArguments.pUserName;
+            }
+        }
+
+        /* Set the MQTT password, as long as it's not empty or NULL. */
+        if( demoArguments.pPassword != NULL )
+        {
+            credentials.passwordSize = strlen( demoArguments.pPassword );
+
+            if( credentials.passwordSize > 0 )
+            {
+                credentials.pPassword = demoArguments.pPassword;
+            }
+        }
 
         /* By default, the credential initializer enables ALPN with AWS IoT,
-         * which only works over port 443. Disable ALPN if another port is
+         * which only works over port 443. Clear that value if another port is
          * used. */
         if( demoArguments.port != 443 )
         {
             credentials.pAlpnProtos = NULL;
         }
 
+        /* Per IANA standard:
+         * https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.xhtml. */
+        if( ( demoArguments.pUserName != NULL ) &&
+            ( demoArguments.awsIotMqttMode == true ) )
+        {
+            credentials.pAlpnProtos = IOT_DEMO_ALPN_FOR_PASSWORD_AUTHENTICATION;
```

Comment:
If the demoArguments.port is 443, and demoArguments.pUserName != NULL  &&  demoArguments.awsIotMqttMode == true , would IOT_DEMO_ALPN_FOR_PASSWORD_AUTHENTICATION for credentials.pAlpnProtos work with AWS IoT MQTT broker?  Is the support for both become available at the same time in Broker?